src/main/java/appeng/fluids/container/ContainerFluidConfigurable.java

```diff
@@ -0,0 +1,117 @@
+
+package appeng.fluids.container;
+
+
+import java.util.Collections;
+import java.util.Map;
+
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.IContainerListener;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+
+import appeng.api.config.Upgrades;
+import appeng.api.implementations.IUpgradeableHost;
+import appeng.api.storage.data.IAEFluidStack;
+import appeng.container.implementations.ContainerUpgradeable;
+import appeng.fluids.helper.FluidSyncHelper;
+import appeng.fluids.util.AEFluidStack;
+import appeng.fluids.util.IAEFluidTank;
+import appeng.util.Platform;
+
+
+public abstract class ContainerFluidConfigurable extends ContainerUpgradeable implements IFluidSyncContainer
+{
+	private FluidSyncHelper sync = null;
+
+	public ContainerFluidConfigurable( InventoryPlayer ip, IUpgradeableHost te )
+	{
+		super( ip, te );
+	}
+
+	public abstract IAEFluidTank getFluidConfigInventory();
+
+	private FluidSyncHelper getSynchHelper()
+	{
+		if( this.sync == null )
+		{
+			this.sync = new FluidSyncHelper( getFluidConfigInventory(), 0 );
+		}
+		return this.sync;
+	}
+
+	@Override
+	protected ItemStack transferStackToContainer( ItemStack input )
+	{
+		FluidStack fs = FluidUtil.getFluidContained( input );
+		if( fs != null )
+		{
+			final IAEFluidTank t = getFluidConfigInventory();
+			final IAEFluidStack stack = AEFluidStack.fromFluidStack( fs );
+			for( int i = 0; i < t.getSlots(); ++i )
+			{
+				if( t.getFluidInSlot( i ) == null && this.isValidForConfig( i, stack ) )
+				{
+					t.setFluidInSlot( i, stack );
+					break;
+				}
+			}
+		}
+		return input;
+	}
+
+	protected boolean isValidForConfig( int slot, IAEFluidStack fs )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "fs". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)